recipes/ginkgo/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+                                         "8.4", "9", "9.1", "9.2", "9.3", "10",
+                                         "10.1"]},
+                             "clang": {"runtime": None, "libcxx": None,
+                                       "version": [
+                                           "3.9", "4.0", "5.0", "6.0", "7.0",
+                                           "7.1", "8", "9", "10", "11"]},
+                             "apple-clang": {"libcxx": None, "version": [
+                                                 "10.0", "11.0", "12.0"]},
+                             "intel": {"version": ["18", "19"]}},
+                "build_type": None, "arch": None}
+    options = {"shared": [True, False], "fPIC": [True, False], "openmp": [
+        True, False], "cuda": [True, False]}
+    default_options = {"shared": False, "fPIC": False,
+                       "openmp": False, "cuda": False}
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["GINKGO_BUILD_TESTS"] = False
+        self._cmake.definitions["GINKGO_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["GINKGO_BUILD_BENCHMARKS"] = False
+        self._cmake.definitions["GINKGO_DEVEL_TOOLS"] = False
+        self._cmake.definitions["GINKGO_BUILD_REFERENCE"] = True
+        self._cmake.definitions["GINKGO_BUILD_OMP"] = self.options.openmp
+        self._cmake.definitions["GINKGO_BUILD_CUDA"] = self.options.cuda
+        self._cmake.definitions["GINKGO_BUILD_HIP"] = False
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        if "fPIC" in self.options:
+            self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.fPIC
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
You should add ... in `package_info`

```py
self.cpp_info.names["pkg_config"] = "the file you deleted"
```

Reply:
Are packages supposed to export their pkg-config information? It sounded to me like all include paths and linker flags are supposed to go through Conan directly.

Reply:
> Are packages supposed to export their pkg-config information?

No

> It sounded to me like all include paths and linker flags are supposed to go through Conan directly.

You are correct!

But conan needs to know which file/name to generate that information in. For that reason we need to _explicitly_ tell it what to do in this case